src/main/java/com/example/qnacomunity/service/GradeService.java

```diff
@@ -0,0 +1,89 @@
+package com.example.qnacomunity.service;
+
+import com.example.qnacomunity.dto.form.GradeForm;
+import com.example.qnacomunity.dto.response.GradeResponse;
+import com.example.qnacomunity.dto.response.MemberResponse;
+import com.example.qnacomunity.entity.Grade;
+import com.example.qnacomunity.exception.CustomException;
+import com.example.qnacomunity.exception.ErrorCode;
+import com.example.qnacomunity.repository.GradeRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class GradeService {
+
+  private final GradeRepository gradeRepository;
+
+  @Transactional
+  public GradeResponse createGrade(GradeForm form) {
+
+    //동일한 이름의 등급이 있는지 체크
+    if (gradeRepository.findByGradeName(form.getGradeName()).isPresent()) {
+      throw new CustomException(ErrorCode.GRADE_NAME_EXIST);
+    }
+
+    //동일한 최소 스코어의 등급이 있는지 체크
+    if (gradeRepository.findByMinScore(form.getMinScore()).isPresent()) {
+      throw new CustomException(ErrorCode.GRADE_SCORE_EXIST);
+    }
+
+    Grade grade = Grade.builder()
+        .gradeName(form.getGradeName())
+        .minScore(form.getMinScore())
+        .build();
+
+    return GradeResponse.from(gradeRepository.save(grade));
+  }
+
+  @Transactional(readOnly = true)
+  public Page<GradeResponse> getAllGrades(Pageable pageable) {
+    return gradeRepository.findAll(pageable).map(GradeResponse::from);
+  }
+
+  @Transactional
+  public GradeResponse updateGrade(Long gradeId, GradeForm form) {
+
+    //동일한 이름의 등급이 있는지 체크
+    if (gradeRepository.findByGradeName(form.getGradeName()).isPresent() &&
+        !gradeRepository.findByGradeName(form.getGradeName()).get().getId().equals(gradeId)
```

Comment:
이렇게 하면 gradeRepository. findByGradeName 가 여러번 불리게 될것 같아요. 
```suggestion
    if(gradeRepository.findByGradeName(form.getGradeName()).map(it -> !it.getId().equals(gradeId)).orElse(false))
```
위와 같은 형태로 chaining 하여 연결하시면 좀더 좋을 것 같구요, 아니면 애초에 query를 할때 Name은 동일하지만 gradeId는 동일하지 않은 데이터만 가지고 와도 좋을 것 같아요